action.yml

```diff
@@ -124,16 +132,19 @@ runs:
 
         # because setup is a Windows GUI app, make it part of a pipeline to make
         # PowerShell wait for it to exit
-        & C:\setup.exe $args | Out-Default
+        & $setupExe $args | Out-Default
       shell: pwsh
 
     - if: ${{ inputs.add-to-path  == 'true' }}
-      run: echo "${{ inputs.install-dir }}\bin" >> $env:GITHUB_PATH
+      run: |
+        $installDir = '${{ inputs.install-dir }}' -ne '' ? '${{ inputs.install-dir }}' : '${{ inputs.work-vol }}\cygwin'
+        echo "$installDir\bin" >> $env:GITHUB_PATH
       shell: pwsh
 
     - run: |
         # run login shell to copy skeleton profile files
-        ${{ inputs.install-dir }}\bin\bash.exe --login
+        $installDir = '${{ inputs.install-dir }}' -ne '' ? '${{ inputs.install-dir }}' : '${{ inputs.work-vol }}\cygwin'
+        & "$installDir\bin\bash.exe" --login
```

Comment:
I think this could just be 'bash --login', since we've just prepended the install-dir to the PATH in the previous step?

Reply:
The previous step runs only `- if: ${{ inputs.add-to-path  == 'true' }}`, which is the default, but can be disabled.